app/src/main/java/com/example/demo/kotlin_practice/functions/MainFunction.kt

```diff
@@ -0,0 +1,120 @@
+package com.example.demo.kotlin_practice.functions
+
+fun main() {
+
+    var objDemo = FuncInterface()
+    println(objDemo.name)
+    objDemo.find()
+
+    var arr = arrayOf(1, 5, 7, 9).sum()
+    println(arr)
+
+    //user-defined function
+    fun add(num1: Int = 5, num2: Int): Int {
+        return num1 + num2
+    }
+
+    var res = add(num2 = 10) //named arg
+    println("Sum: $res")
+
+    fun display(name: String) {
+        if (name != null) {
+            println("Hello $name")
+        } else {
+            println("hiii")
+        }
+    }
+    display("vidhi")
+
+    //single line function
+    fun mul(x: Int) = x * 2
+    println(mul(5))
+
+    //vararg
+    fun addNum(vararg num: Int): Int = num.sum()
+    println(addNum(1, 5, 7, 9))
+
+    fun addNames(vararg num: Int, name: String) {
+        println("Sum: ${num.sum()} and name: $name")
+    }
+    addNames(2, 4, 9, name = "vidhi")
+
+    //infix call
+    val obj = Demo()
+    println(obj display 4)
+
+    //recursion - fact
+    val ansOf = fact(4)
+    println("Factorial: $ansOf")
+
+    //lambdas
+    println(sum2(5, 7))
+
+    //shorthand lambda
+    println(arrNum.filter { it > 0 })
+
+    //anonymous functions
+    println(an(5, 6))
+    println(mul(5, 6))
+
+    //inline
+    println(foo { println("inline fun") })
+
+    //cross-inline
+    inDemo({ println("First inline ") },
+        { println("Second inline") })
```

Comment:
do not line break until line can not fit into around 120 characters horizontally